### PR TITLE
More info on front-end extensions

### PR DESCRIPTION
**Pull request by jcb91**
in docs/extending/frontend_extensions:
* Add some description of actions
* beautify javascript examples
* fix inline literals, which in RST start and end with ``, not just `
* fix some typos & misspellings, some improvements to phrasing

### CODE REVIEW

File: docs/source/extending/frontend_extensions.rst [hunk 1]
```diff
@@ -5,125 +5,210 @@ This describes the basic steps to write a JavaScript extension for the Jupyter
...
+of what is available:
 
 .. code:: javascript
 
-    Object.keys(IPython.actions._actions)
+    Object.keys(require('base/js/namespace').actions._actions);
 
-In this example, we changed keyboard shortcuts in **command mode**; you can also
-customise keyboard shortcuts in edit mode.
+In this example, we changed the keyboard shortcut in the **command mode**; you
```
Comment (takluyver) [inline docs/source/extending/frontend_extensions.rst]: '...changed *a* keyboard shortcut in command mode' (both instances of 'the' read wrong here)

**Comment by takluyver** [comment]
Thanks, I think this looks good.

In some of our docs, we set `default_role = 'literal'` in the config, so that we can do inline code using single backticks, as we're used to in Markdown. But we haven't actually done that in the notebook docs yet, so you're right to double up the backticks.

File: docs/source/extending/frontend_extensions.rst (hunk 1 above)
Comment (jcb91) [inline docs/source/extending/frontend_extensions.rst]: good spot, done!

**Comment by takluyver** [comment]
Thanks!

**Pull request merged**